### ISSUE TITLE
[a11y]: RadioButtonGroup <fieldset> element does not have a <legend>

### ISSUE THREAD

**Issue by curtispd** [carbon-design-system/carbon-website#3275]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

10.58.4

### React version

16.14.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker, Latest Deployment, IBM Accessibility

### Description

Accessibility findings
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/2791009/202530345-b369b352-a961-4800-b708-4b0754d4a55e.png">

<fieldset> elements must have a single, non-empty <legend> as a label
     <fieldset> element does not have a <legend>

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example

https://carbondesignsystem.com/components/radio-button/usage/

### Steps to reproduce

1. Open https://carbondesignsystem.com/components/radio-button/usage/
2. Run accessibility scan

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by curtispd** [comment]
You're right. I don't see the inner fieldset on your link. I'm using Carbon 10 on my page and it shows the same issue as the website example.

**Comment by tw15egan** [comment]
The [latest release of v10](https://v7-react.carbondesignsystem.com/iframe.html?id=components-radiobutton--default&args=&viewMode=story) does not seem to have this extra `fieldset` element...

Oh, I think I see the issue (on the website, at least). The `RadioButtonGroup` prop should be `legendText` not `legend`. It also has a redundant `FormGroup` wrapped around it. This could probably be removed.

**Comment by curtispd** [comment]
It seems the component I'm using (which uses RadioButtonGroup) was done following the website example so it has the legendText on the wrapping FormGroup but not on the RadioButtonGroup. So it's an issue in docs/usage rather than the component itself. Thanks !

**Comment by tw15egan** [comment]
Correct, @curtispd. I've put up a PR to fix this on the website if you want to check it out. Sorry for the confusion 😅
